packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -276,13 +359,49 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
       mounted.current = true;
       return;
     }
-    const currentValue = optionsRef.current[selectedIndex]?.value;
+    if (isMultipleEnabled(selectedIndex, multiple)) {
+      const currentOptions = selectedOptions as SelectOption<T>[];
+      if (
+        !isSingleOnChange(onChangeProp.current, multiple) &&
+        currentOptions.length > 0
+      ) {
+        currentOptions.map((option) =>
```

Comment:
This part is not nice, since it will trigger `onChange` for every already selected value, no matter that only one item was selected.

Reply:
Should we give users back single changed value or full array? 
`onChange => ((value: T, event: SelectValueChangeEvent) => void)` (multiselect does this)
or
`onChange => ((values: T[]) => void)`

Reply:
In `Select` in calls `onChange` only one time with the changed item, whether it was added or removed. And now when you select or deselect an item, `onChange` is being called on each selected item, not just the newest one.